src/molecule_docker/test/scenarios/env-substitution/molecule/default/molecule.yml

```diff
@@ -0,0 +1,10 @@
+---
+driver:
+  name: docker
+platforms:
+  - name: instance-${DOES_NOT_EXIST:-local}
+    image: env-substitution
```

Comment:
Could you make a slight change?

```txt
- name: instance
  image: ${MOLECULE_ROLE_IMAGE:-centos:7}
```

Reply:
What do you want to achieve by renaming the image? When setting this to centos:7 it would still build the Debian dockerfile if I'm not mistaken.

Reply:
It's fun to use `debian` as base image.

I just want to highlight `image: ${MOLECULE_ROLE_IMAGE:-SOME_IMAGE:SOME_TAG}`.

No matter `Dockerfile.j2` exists, if `${}` is not rendered correctly, the issue may occur.

Reply:
IMO, the root cause of #184 is that the `molecule.yml` is overwritten by mistake.
So I think the minimum test case is
1. No need to add a `Dockerfile.j2` in test case
2. Set an environment variable in `image` in `molecule.yml`
3. Set `pre_build_image: true` in `molecule.yml`

Let me know what you think.

Reply:
I will need the dockerfile so the context is copied for the image. That is the problem. If I don't use the dockerfile there is no context copy happening and molecule.yml won't be overwritten (which it shouldn't anymore anyways).

Reply:
`pre_build_image` is set to `false` by default, so the `Synchronization the context` would run by default as well.

I think it's not the `Dockefile.j2` but `pre_build_image: false`.

This is @juju4 described in this comment. https://github.com/ansible-community/molecule-docker/issues/184#issuecomment-1272318154

Only if `pre_build_image` is set to `true` explicitly, the issue would be evaded.